enclave/seal.c

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/internal/crypto/gcm.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/safecrt.h>
+#include <openenclave/internal/entropy.h>
+
+static const uint8_t _iv[12] = { 0 };
```

Comment:
`_iv` is a constant of 12 zero bytes. Because of that, **no seal key** can be used more than once. This is the existing behavior of Intel SGX SDK and is carried through here for interoperability.

The rationale behind this is that, as [NIST.SP300-38D](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf) stipulates, any given key shall not be used in more than 2<sup>32</sup> invocations. But how can `oe_seal()` enforce that? The best way is probably to use a fresh key on each invocation. Given a fresh key, there's no need to use distinct/random IV. Therefore, a constant `_iv` of zero is used here for simplicity.